src/ProjectTemplates/test/ProjectTemplates.Tests.Common/Infrastructure/GenerateTestProps.targets

```
@@ -0,0 +1,94 @@
+    DependsOnTargets="PrepareForTest"
+    Condition="$(DesignTimeBuild) != true">
+    <PropertyGroup>
+      <PropsProperties>
+        RestoreSources=$([MSBuild]::Escape("$(RestoreSources);$(ArtifactsShippingPackagesDir);$(ArtifactsNonShippingPackagesDir)"));
+        MicrosoftNETCoreAppPackageVersion=$(MicrosoftNETCoreAppPackageVersion);
+        MicrosoftNETCorePlatformsPackageVersion=$(MicrosoftNETCorePlatformsPackageVersion);
+        MicrosoftNETSdkRazorPackageVersion=$(MicrosoftNETSdkRazorPackageVersion);
+        MicrosoftAspNetCoreAppPackageVersion=$(SharedFxVersion);
+        SupportedRuntimeIdentifiers=$(SupportedRuntimeIdentifiers);
+      </PropsProperties>
+    </PropertyGroup>
+
+    <Sdk_GenerateFileFromTemplate
+      TemplateFile="$(MSBuildThisFileDirectory)\TemplateTests.props.in"
+      Properties="$(PropsProperties)"
+      OutputPath="$(TestTemplateTestsProps)" />
+  </Target>
+
+   <Target Name="PrepareForTest" BeforeTargets="CoreCompile" Condition="$(DesignTimeBuild) != true">
+    <PropertyGroup>
+      <TestTemplateCreationFolder>$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)$([MSBuild]::EnsureTrailingSlash('$(OutputPath)$(TestTemplateCreationFolder)'))'))</TestTemplateCreationFolder>
+      <TestTemplateTestsProps>$(TestTemplateCreationFolder)$(TestTemplateTestsProps)</TestTemplateTestsProps>
+      <CustomTemplateHivePath>$(TestTemplateCreationFolder)\Hives\$([System.Guid]::NewGuid())\.templateengine</CustomTemplateHivePath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
+        <_Parameter1>ArtifactsShippingPackagesDir</_Parameter1>
+        <_Parameter2>$(ArtifactsShippingPackagesDir)</_Parameter2>
+      </AssemblyAttribute>
+
+      <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
+        <_Parameter1>ArtifactsNonShippingPackagesDir</_Parameter1>
+        <_Parameter2>$(ArtifactsNonShippingPackagesDir)</_Parameter2>
+      </AssemblyAttribute>
+
+      <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
+        <_Parameter1>TestTemplateCreationFolder</_Parameter1>
+        <_Parameter2>$(TestTemplateCreationFolder)</_Parameter2>
+      </AssemblyAttribute>
+      <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
+        <_Parameter1>CustomTemplateHivePath</_Parameter1>
+        <_Parameter2>$(CustomTemplateHivePath)</_Parameter2>
+      </AssemblyAttribute>
+    </ItemGroup>
+
+    <Message Importance="high" Text="Preparing environment for tests" />
+    <!-- Remove the template creation folders and the package-restore folders to ensure that when we run the tests we don't
+         get cached results and changes show up.
+    -->
+
+    <ItemGroup>
+      <_ExistingFilesFromLastRun Include="$(TestTemplateCreationFolder)**\*" />
+    </ItemGroup>
+
+    <Delete Files="@(_ExistingFilesFromLastRun)" ContinueOnError="true" />
```

Personal preference is to use the enum expression for ContinueOnError since there are actually 3 behaviors to choose from. `true` is equivalent to `WarnAndContinue`.
```suggestion
    <Delete Files="@(_ExistingFilesFromLastRun)" ContinueOnError="WarnAndContinue " />
```